.github/workflows/testCoverage.yml

```diff
@@ -17,20 +30,87 @@ jobs:
+        
+      - name: Run the tests in macos
+        if: matrix.os == 'macos-latest'
+        run: cd ./repo1/packages/react-charting && sed -i '' 's/PROD/TEST/g' ./config/tests.js && yarn jest --coverage --coverageDirectory ./coverage/${{matrix.os}} --verbose --coverageReporters=html --coverageReporters=text > coverageReport.txt
+        
+      - name: Run the tests in ubuntu
+        if: matrix.os == 'ubuntu-latest'    
+        run: cd ./repo1/packages/react-charting && sed -i 's/PROD/TEST/g' ./config/tests.js && yarn jest --coverage --coverageDirectory ./coverage/${{matrix.os}} --verbose --coverageReporters=html --coverageReporters=text > coverageReport.txt        
+
+      - name: Generate coverage file name in windows
+        if: matrix.os == 'windows-latest'          
+        run: |
+            $NOW=& Get-Date -format yyyy-MM-dd
+            echo "COVERAGE_FILENAME=test_coverage_${{ matrix.os }}_$NOW" >> $env:GITHUB_ENV
+
+      - name: Generate coverage file name
+        run: |          
+            echo "COVERAGE_FILENAME=test_coverage_${{ matrix.os }}_$(date +'%Y-%m-%d_%H-%M-%S')" >> $GITHUB_ENV
+
+      - name: Save coverage folder
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{env.COVERAGE_FILENAME}}
+          path: ./repo1/packages/react-charting/coverage/${{matrix.os}}
+
+      - name: Save coverage file
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{env.COVERAGE_FILENAME}}_report
+          path: ./repo1/packages/react-charting/coverageReport.txt
+
+      - name: Extract the coverage summary table
+        if: matrix.os != 'windows-latest'
+        run: |
+            cd ./repo1/packages/react-charting
+            file_contents=$(cat coverageReport.txt)
+            table=$(echo "$file_contents" | awk '/----/,/^$/' | sed '1d;$d')
+            echo "$table"
+
+      - name: Extract the coverage summary table in windows
+        if: matrix.os == 'windows-latest'
+        run: |
+            cd ./repo1/packages/react-charting
+            $fileContents = Get-Content -Raw -Path "coverageReport.txt"
+            $tableRegex = "(?ms)----.*?^$"
+            $table = [regex]::Match($fileContents, $tableRegex)
+            echo "$table"
+
+      - name: Setup Pages
+        if: matrix.os == 'windows-latest' 
+        uses: actions/configure-pages@v4
+        
+      - name: Upload artifact
+        if: matrix.os == 'windows-latest'
+        uses: actions/upload-pages-artifact@v3
+        with:
+          # Upload entire repository
+          path: './repo1/packages/react-charting/coverage/${{matrix.os}}'
+          
+      - name: Deploy to GitHub Pages
+        id: deployment
+        if: matrix.os == 'windows-latest'
+        uses: actions/deploy-pages@v4
```

Comment:
We dont want to deploy coverage report on the main github pages slot. We plan to publish the contributor guide to it. Can the coverage report be published to a sub path or alternative github page.

Reply:
Updated accordingly to accommodate both the contributor guide and the test coverage reports in the same site.